client/components/settingPage/index.js

```diff
@@ -1,9 +1,250 @@
-import React from 'react';
+import axios from 'axios';
+import { Alert } from 'react-bootstrap';
+import { Button } from 'react-bootstrap';
+import React, { useState, useRef, useContext } from 'react';
+import { firebaseStorage } from '../../../Docs/firebase-config/firebase';
+
+import './setting.css';
+
+export const SettingPage = ({ user }) => {
+  const [curTab, setCurTab] = useState('profile');
+  const [msg, setMsg] = useState('');
+  const [error, setError] = useState('');
+  const [usrEmail, setUsrEmail] = useState('');
+  const [usrName, setUsrName] = useState('');
+  const [usrOldPwd, setUsrOldPwd] = useState('');
+  const [usrNewPwd, setUsrNewPwd] = useState('');
+  const [ccNum, setCcNum] = useState(null);
+  const [ccHolder, setCcHolder] = useState(null);
+  const [ccDate, setCcDate] = useState(null);
+  const [ccCVC, setCcCVC] = useState(null);
+
+  const fileRef = useRef();
+
+  const newImg = () => {
+    document.getElementById('hiddenImg').click();
```

Comment:
It somewhat concerns me when I see you adding 'hiddenImg', but disregarding that I don't think using basic HTML DOM functions is good in react. I believe using a ref is the best way to do it.

Reply:
hmm okay. Was making an image act as a file uploader, so I hide the file uploader behind the image